access/views.py

```diff
@@ -39,10 +46,125 @@ def index(request):
         })
     return render(request, 'access/ready.html', {
         "courses": courses,
-        "manager": 'gitmanager' in settings.INSTALLED_APPS,
     })
 
 
+def publish(request):
+    """
+    Move a course from store to the main folder
+    """
+    if request.method != "POST":
+        raise Http404()
+
+    if "course_id" not in request.POST:
+        return HttpResponse("Missing course_id", status=400)
+
+    course_id = request.POST["course_id"]
+
+    if not request.auth.permissions.instances.has(Permission.WRITE, id=int(course_id)):
+        return HttpResponse(status=401)
+
+    root_dir = Path(settings.COURSES_PATH)
+    store_root_dir = Path(settings.COURSE_STORE)
+    course_path = root_dir / course_id
+    store_course_path = store_root_dir / course_id
+    version_id_path = root_dir / (course_id + ".version")
+    store_version_id_path = store_root_dir / (course_id + ".version")
+
+    store_version_id = None
+    if store_version_id_path.exists():
+        with open(store_version_id_path) as f:
+            store_version_id = f.read()
```

Comment:
```suggestion
    try:
        with open(store_version_id_path) as f:
            store_version_id = f.read()
    except OSError:
        # how should the error be handled?
        return HttpResponse("Failed")
```
File IO in this case should absolutely use try-except instead of assuming that the file can not be removed between `exists()` and `read()`.